### ISSUE TITLE
Handling memory fatal errors

### ISSUE THREAD

**Issue by mstrychalski** [juriansluiman/SlmQueueDoctrine#142]
Sometimes it can happen, that script will allocate too much memory (eg. pdf generation) and then it throws uncatchable fatal error.

SlmQueueDoctrine recovers those jobs and runs it over and over, not firing up any event, and I can't find any way to handle those errors. I can set executionTime for recover, and they will not be recovered, but those jobs are still left unprocessed in database, and they just die silently.

What I need to do, is to not recover jobs, and set some things, eg. using Strategy. When memory is out, I can't do anything besides registering new register_shutdown_function, rising memory limit and do what I want. But this is reduntant, because I am already using Strategies to catch errors, send them to raygun or log them. 

In this case nothing seems to work, script is stopped, slmqueue is trying to recover it, and it is stopped again. No event fired.

Is there any way to mark those jobs as finished with error, unrecoverable?

**Comment by basz** [comment]
> SlmQueueDoctrine recovers those jobs and runs it over and over, not firing up any event, and I can't find any way to handle those errors.

I don't think this is correct. When a Job is processed (status goes to 2) and the workers dies for whatever reason the job is left inside the table with status 2 as if it is still running. SlmQueueDoctrine doesn't do anything special unless you explicitly throw a [ReleasableException](https://github.com/juriansluiman/SlmQueueDoctrine/blob/master/src/Worker/DoctrineWorker.php#L32) or [BuryableException](https://github.com/juriansluiman/SlmQueueDoctrine/blob/master/src/Worker/DoctrineWorker.php#L36) inside your job itself.

If you have thrown a ReleasableException the Job would be marked as pending again. Are you sure you aren't doing this...

**Comment by basz** [comment]
No in php it is not possible to recovery from fataal errors. In fact the is a list of Error types you can not recovery from

> The following error types cannot be handled with a user defined function: E_ERROR, E_PARSE, E_CORE_ERROR, E_CORE_WARNING, E_COMPILE_ERROR, E_COMPILE_WARNING, and most of E_STRICT raised in the file where set_error_handler() is called.
